betree/pmdk/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+use std::sync::{Arc, Mutex};
+use std::os::raw::c_void;
+use std::ptr::NonNull;
+
+#[derive(Debug)]
+//pub struct ptr_to_pmem(Arc<Mutex<*mut c_void>>);
+pub struct ptr_to_pmem(Arc<*mut c_void>);
+
+unsafe impl Send for ptr_to_pmem {}
+unsafe impl Sync for ptr_to_pmem {}
+
+pub mod libpmem {
+    use super::*;
+
+    pub fn pmem_file_create(filepath : &str, len: u64, mapped_len : &mut u64, is_pmem : &mut i32) -> Option<ptr_to_pmem> {
+        unsafe {
+            let mut ptr = pmem_map_file(filepath.as_ptr() as *const i8,
+                                     len,
+                                     (PMEM_FILE_CREATE|PMEM_FILE_EXCL) as i32,
+                                     0666,
+                                     mapped_len,
+                                     is_pmem);
+
+            match NonNull::new(ptr) {
+                //Some(value) => Some(ptr_to_pmem( Arc::new(Mutex::new(ptr)))),
+                Some(value) => Some(ptr_to_pmem( Arc::new(ptr))),
+                None => None
+            }
+        }
```

Comment:
It would be best to confine `unsafe` usage to the minimum required here. Only the `pmem_map_file` is unsafe.